### DISCUSSION TITLE
Dropdown WITHOUT Javascript

### DISCUSSION THREAD

**Discussion by shehi** [twbs/bootstrap#36651] [Q&A] [answered]
Using Bootstrap 5.2-beta.1 here.

I am really confused. Docs say one can create Dropdown without Javascript, using `data-bs-toggle` attrib only. That doesn't work. So question is: Is it even possible to create a Dropdown with JS disabled, or at the very least, not including Bootstrap's JS lib?

**Comment by mdo** [comment, marked as answer]
Where do our docs say that? Dropdowns require JS to toggle their menus. You don't need to _write_ JavaScript to use them, just HTML with our data attributes and the proper script tags.

**Reply by julien-deramond** [reply]
I kinda understand what could be confusing maybe in https://getbootstrap.com/docs/5.2/components/dropdowns/#usage where it is said "Via data attributes or JavaScript" (+ 2 separate sections). One could maybe understand that "Via data attributes" doesn't need JS at all.

**Reply by shehi** [reply]
> I kinda understand what could be confused maybe in https://getbootstrap.com/docs/5.2/components/dropdowns/#usage where it is said "Via data attributes or JavaScript" (+ 2 separate sections). One could maybe understand that "Via data attributes" doesn't need JS at all.

Exactly. Would be great if Docs actually emphasize that JS usage is as important as the usage of `data-bs-toggle` attribute.

I wish I could mark @julien-deramond 's reply as an Accepted Answer, but it's just a reply :) Thank you guys, this sucks, lol. I wish we had CSS-only version of these dropdowns, as it is easy to make them (maybe not with all features the current implementation supports, but still, it is). I use React on NextJs, and `react-bootstrap` is extremely opinionated (sticking `href` fields into their `Dropdown.Item` components, which I don't need). Your Js needs extra work to be used in such environments.

**Reply by julien-deramond** [reply]
Thanks @shehi for your feedback.
I'll try to think about something to improve the documentation to make it clearer regarding the point mentioned in this discussion.
Regarding the dropdown without JS, I'll try to check in the history of the project and to think about something just to see if it would be considerable in Bootstrap. That could maybe be interesting in terms of eco-design to have a "light" version of the dropdowns.

**Reply by KelSolaar** [reply]
Hi,

That would be a good feature, I have one user venting about our website being unusable without JS: https://github.com/colour-science/colour-science.org/issues/2

It mainly comes down to our nav not working without it.

Cheers,

Thomas

**Comment by KOLANICH** [comment]
I'm that user and I mean "no JS" that widgets should be implemented in pure HTML + CSS and work without JS. Because JS and browser API are used for malicious purposes too a lot and browser vendors cannot really fix it because

1. they can break too lot of benign libs
2. malicious actors can discriminate against users using the browsers whose developers have implemented the needed protections in order to cause the users to change the browsers, the companies developing those browsers revenues go down and the companies got extinct (bankrupt)

https://github.com/you-dont-need/You-Dont-Need-JavaScript may be helpful for the quest of getting rid of JS.
